qt/applications/workbench/workbench/plotting/figureerrorsmanager.py

```diff
@@ -0,0 +1,343 @@
+
+        # iterates over all lines, in order to
+        # toggle error bar on lines that have errors already
+        data_lines = self.get_data_lines(ax)
+        for index, container in enumerate(ax.containers):
+            data_line = container[0]
+
+            # extract the line reference from the container
+            if data_line.get_label() == MantidAxes.MPL_NOLEGEND:
+                self._toggle_error_bar_for(data_lines.index(data_line),
+                                           errorbar_container=container,
+                                           make_visible=make_visible)
+
+        # errors can't be added dynamically while working with axes that aren't supported
+        # also if there is no errobar container present, and the state if being forced to
+        # make_visible == False, then avoid doing any more work
+        if not self._supported_ax(ax) or not make_visible:
+            return
+
+        # Iterate over all lines to add new error bars.
+        # Each added errorbar changes the lines list, making the iteration not just 1..n.
+        # The line is always removed from pos `index`,
+        # and errors are added at the end of the list, the next element is moved 1 position up.
+        # Therefore the index should only be increased
+        # if the line it is looking for already has errors plotted
+        for index, data_line in enumerate(data_lines):
+            if data_line.get_label() != MantidAxes.MPL_NOLEGEND:
+                self._add_errorbar_for(index, data_line, make_visible)
+
+    def _toggle_error_bar_for(self, index, errorbar_container, make_visible=None):
+        """
+        :param make_visible: Forces this visibility state on all lines.
+        :type make_visible: bool
+        :param index: The index in the creation_args
+        :type index: int
+        :return:
+        """
+
+        ax = self.canvas.figure.axes[0]  # type: matplotlib.axes.Axes
+
+        # Currently the X errors are not touched at all.
+        # Only operations on the Y errors are done.
+        y_err_index = self._find_yerr_index(errorbar_container)
+
+        # if the yerr is plotted on the plot
+        if y_err_index is not None:
+            error_line = errorbar_container.lines[2][y_err_index]
+            # the container has the errors there
+            # if a force_state is passed, use that, otherwise invert the current state
+            new_state = not error_line.get_visible() if make_visible is None else make_visible
+
+            # updates the creation args state if the plot would be saved out
+            if self._supported_ax(ax):
+                creation_args = ax.creation_args
+                if MantidAxKwargs.POST_CREATION_ARGS in creation_args[index]:
+                    creation_args[index][MantidAxKwargs.POST_CREATION_ARGS][MantidAxKwargs.ERRORS_VISIBLE] = new_state
+                else:
+                    creation_args[index][MantidAxKwargs] = {MantidAxKwargs.ERRORS_VISIBLE: new_state}
```

Comment:
Is this missing a `.POST_CREATION_ARGS`?

Reply:
Yes, good spot